miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/permission/MiniAppCustomPermissionCache.kt

```diff
@@ -44,24 +80,29 @@ internal class MiniAppCustomPermissionCache(context: Context) {
     fun storePermissions(
         miniAppCustomPermission: MiniAppCustomPermission
     ) {
-        val cached = readPermissions(miniAppCustomPermission.miniAppId).pairValues
         val supplied = miniAppCustomPermission.pairValues.toMutableList()
 
         // Remove any unknown permission parameter from HostApp.
         supplied.removeAll { (first) ->
             first.type == MiniAppCustomPermissionType.UNKNOWN.type
         }
 
+        val miniAppId = miniAppCustomPermission.miniAppId
+        val allPermissions = prepareAllPermissionsToStore(miniAppId, supplied)
+        applyStoringPermissions(MiniAppCustomPermission(miniAppId, allPermissions))
+    }
+
+    /**
+     * Apply to be saved the grant results as json string to SharedPreferences.
+     * @param [miniAppCustomPermission] an object to contain the results per MiniApp.
+     */
+    @VisibleForTesting
```

Comment:
Is it internal function or public one? You can remove this annotation if it can be public, otherwise should declare it as internal.

The same for other functions in this file.

Reply:
We don't need to make each function `internal` as the entire class is already `internal`.

Reply:
It makes sense but if you don't declare the function as internal, it means that other components in our sdk can call it freely. The internal here actually express that you want your function to be private but you make it internal for testing.

Reply:
I have added a `WantedPrivateButTesting` annotation to mark this kind of function to express the usage.

Reply:
Wise solution.

Reply:
I tested and `WantedPrivateButTesting` does not behave like `VisibleForTesting`. It does not warn the developer to not use method publicly.

Reply:
I guess, the annotation required to be stored to behave like `@VisibleForTesting`. So far, I am using `AnnotationRetention.SOURCE` policy, probably this is a reason. I haven't tried other policies yet. 

By the way, I just found that using `@VisibleForTesting(otherwise = VisibleForTesting.NONE)` also may resolve our needs according to the [official guide](https://developer.android.com/studio/write/annotations). It's still visible to other classes, but by using it, I can have warning in other classes.  

e.g. 

![Screen Shot 2020-11-13 at 15 08 29](https://user-images.githubusercontent.com/66930373/99036716-85e38e80-25c5-11eb-855f-6d83d8f42ae5.png)

@minh-rakuten What do you think?

Reply:
I think you don't need to have own annotation. Just use `@VisibleForTesting` and declare `internal` for private functions.

Reply:
Ok, I'll use `@VisibleForTesting(otherwise = VisibleForTesting.NONE)` as officially recommended.